src/js/actions/libraries.js

```diff
@@ -821,26 +829,50 @@ define(function (require, exports) {
-                                return this.transfer(layerActions._getLayerIDsForDocumentID, currentDocument.id);
-                            })
-                            .then(function (nextDocumentIDS) {
-                                var nextLayerIDs = nextDocumentIDS.layerIDs,
-                                    existingLayerIDs = currentDocument.layers.index.toArray(),
-                                    newLayerIDs = _.difference(nextLayerIDs, existingLayerIDs).reverse();
-
-                                return this.transfer(layerActions.addLayers, currentDocument, newLayerIDs, true, false);
-                            });
+                            return this.transfer(layerActions._getLayerIDsForDocumentID, currentDocument.id)
+                                .bind(this)
+                                .then(function (nextDocumentIDS) {
+                                    var nextLayerIDs = nextDocumentIDS.layerIDs,
+                                        existingLayerIDs = currentDocument.layers.index.toArray(),
+                                        newLayerIDs = _.difference(nextLayerIDs, existingLayerIDs).reverse();
+
+                                    return this.transfer(layerActions.addLayers, currentDocument,
+                                        newLayerIDs, true, false);
+                                });
+                        }
+                    })
+                    .catch(function () {
+                        if (!hasAlt) {
+                            return this.transfer(handleCompletePlacingGraphic);
                         }
-                        
-                        return placeElementPromise;
                     });
             });
     };
     createLayerFromElement.reads = [locks.CC_LIBRARIES, locks.JS_DOC, locks.JS_UI, locks.JS_APP];
     createLayerFromElement.writes = [locks.PS_DOC];
     createLayerFromElement.transfers = [layerActions._getLayerIDsForDocumentID, layerActions.addLayers,
-        policyActions.addKeydownPolicy, policyActions.removeKeyboardPolicies];
+        policyActions.suspendAllPolicies, "libraries.handleCompletePlacingGraphic"];
+        
+    /**
+     * This event will be triggered when the user confirm or cancel the new layer 
+     * created from createLayerFromElement
+     * 
+     * @return {Promise}
+     */
+    var handleCompletePlacingGraphic = function () {
+        return this.dispatchAsync(events.libraries.PLACE_GRAPHIC_UPDATED, { isPlacing: false })
+            .bind(this)
+            .then(function () {
+                var policyStore = this.flux.store("policy");
+                
+                if (policyStore.areAllSuspended) {
+                    return this.transfer(policyActions.restoreAllPolicies);
+                }
+            });
+    };
+    handleCompletePlacingGraphic.reads = [locks.JS_APP];
+    handleCompletePlacingGraphic.writes = [];
```

Comment:
does PLACE_GRAPHIC_UPDATED not write to an of our stores?  just *seems* like there should be a write lock.  but maybe not